### ISSUE TITLE
Scroll too sensitive on trackpad, and proposition to add a 'scroll Sensitivity' option

### ISSUE THREAD

**Issue by eelfonik** [alvarotrigo/pagePiling.js#104]
First thanks for such a great plugin!

I've used fullPage.js before and now pagePiling.js, one common issue is the scroll detection is too sensitive, especially when you use trackpad on laptop: a slight two finger scroll (ou even just two finger touch) on trackpad will cause the sections to change.

I've found in source code the following to perform a mouseWheel detection & scroll
`/**
         * Detecting mousewheel scrolling
         *
         * http://blogs.sitepointstatic.com/examples/tech/mouse-wheel/index.html
         * http://www.sitepoint.com/html5-javascript-mouse-wheel/
         */
        var prevTime = new Date().getTime();

        function MouseWheelHandler(e) {
         	var curTime = new Date().getTime();
         	// cross-browser wheel delta
             e = e || window.event;
             var value = e.wheelDelta || -e.deltaY || -e.detail;
             var delta = Math.max(-1, Math.min(1, value));

             var horizontalDetection = typeof e.wheelDeltaX !== 'undefined' || typeof e.deltaX !== 'undefined';
             var isScrollingVertically = (Math.abs(e.wheelDeltaX) < Math.abs(e.wheelDelta)) || (Math.abs(e.deltaX ) < Math.abs(e.deltaY) || !horizontalDetection);

             //Limiting the array to 150 (lets not waste memory!)
              if(scrollings.length > 149){
                 scrollings.shift();
              }

            //keeping record of the previous scrollings
            scrollings.push(Math.abs(value));

            //time difference between the last scroll and the current one
             var timeDiff = curTime-prevTime;
             prevTime = curTime;

             //haven't they scrolled in a while?
             //(enough to be consider a different scrolling action to scroll another section)
             if(timeDiff > 200){
                 //emptying the array, we dont care about old scrollings for our averages
                 scrollings = [];
             }

            if(!isMoving()){
                var activeSection = $('.pp-section.active');
                var scrollable = isScrollable(activeSection);
                
                //keeping record of the previous scrollings
                scrollings.push(Math.abs(value));

                var averageEnd = getAverage(scrollings, 10);
                var averageMiddle = getAverage(scrollings, 70);
                var isAccelerating = averageEnd >= averageMiddle;

                if(isAccelerating && isScrollingVertically){
	                //scrolling down?
	                 if (delta < 0) {
	                    scrolling('down', scrollable);

	                //scrolling up?
	                 }else if(delta>0){
	                     scrolling('up', scrollable);
	                 }
	            }

                return false;
            }
         }`

However, the delta value here is varied with different browsers, and sometimes (for example on Mac trackpad or mouse) it's very sensitive with the slightest move. And the isAccelerating boolean in most cases always returns true ( especially if the scrollings array is shorter ).

I tried to make a workaround with a normalizeWheel function taken from here:
https://github.com/facebook/fixed-data-table/blob/master/src/vendor_upstream/dom/normalizeWheel.js
From which I can get the pixel units of mouse/wheel event, and specify only to scroll when the absolute value of pY is larger than a certain number, not purely based on if the delta is positive or negative.

The problem of this workaround is, as stated in the normalizeWheel function, that the pixel units might have "crazy differences between browsers", any thoughts?

**Comment by alvarotrigo** [comment]
Not an easy thing to do and, as far as I know, nobody has ever been able to solve it. 
[There's another library](https://github.com/d4nyll/lethargy) which tries to solve it, but as far as I remember I didn't get the results I expected. I'll have to take a look at it again with more calm.

**Comment by jasperSpeicher** [comment]
Alvaro, your software is great. Thank you!

I am curious if there is any new thought on this. I am noticing that with a magic mouse, there is a good chance that the user will try to click a button, but instead a section transition is initiated.

**Comment by alvarotrigo** [comment]
> I am noticing that with a magic mouse, there is a good chance that the user will try to click a button, but instead a section transition is initiated.

Why? How? 
A click requires a single finger. Scrolling two.

**Comment by jasperSpeicher** [comment]
No, actually on the mouse I am testing with, scrolling happens with one finger: https://en.wikipedia.org/wiki/Magic_Mouse

And there are no settings to change that as far as I can see...

![image](https://cloud.githubusercontent.com/assets/930901/23775854/9ed0fc3a-04df-11e7-8770-f3b09804d818.png)

![image](https://cloud.githubusercontent.com/assets/930901/23775865/b0d5b6a0-04df-11e7-8dcc-f1c71aeb222e.png)

**Comment by alvarotrigo** [comment]
Right. Sorry, I though you were speaking about the trackpad.

**Comment by plucile** [comment]
Thanks for your plugin @alvarotrigo, it's great! 
Any update on this issue however? I'm having the same problem using a magic mouse...
Thank you

**Comment by alvarotrigo** [comment]
@plucile assume there's no updates until I post something :)
